code/ModelAdmin.php

```diff
@@ -183,6 +185,55 @@ public function Link($action = null)
         return parent::Link($action);
     }
 
+    /**
+     * Get the link for the tab of a DataObject subclass managed by this ModelAdmin.
+     *
+     * Note that if the modelClass is managed on multiple tabs and you want a link for a
+     * specific tab you should use {@link getLinkForModelTab} instead.
+     *
+     * @throws InvalidArgumentException if $modelClass is not managed by this ModelAdmin.
+     */
+    public function getLinkForModelClass(string $modelClass): string
+    {
+        if (!$this->isManagedModel($modelClass)) {
+            throw new InvalidArgumentException("$modelClass isn't managed by this ModelAdmin.");
+        }
+        return $this->getLinkForModelTab($this->getModelTabForModelClass($modelClass));
+    }
+
+    /**
+     * Get the link for a specific tab on this ModelAdmin.
+     *
+     * If you do not have multiple tabs for a given class you should use {@link getLinkForModelClass}
+     * instead.
+     *
+     * @throws InvalidArgumentException if $modelTab is not a tab on this ModelAdmin.
+     */
+    public function getLinkForModelTab(string $modelTab): string
+    {
+        // Don't use isManagedModel here because a subclass of a managed model may not have its own tab.
+        if (!array_key_exists($modelTab, $this->getManagedModels())) {
+            throw new InvalidArgumentException("$modelTab isn't a tab on this ModelAdmin.");
+        }
+        return $this->Link($this->sanitiseClassName($modelTab));
+    }
+
+    /**
+     * Get the link for editing an object inside this ModelAdmin.
+     */
+    public function getEditLinkForDataObject(DataObject $obj): string
+    {
+        $link = static::join_links(
+            $this->getLinkForModelClass($obj->ClassName),
+            'EditForm/field/',
+            $this->sanitiseClassName($this->getModelTabForModelClass($obj->ClassName)),
+            'item',
+            $obj->ID
+        );
+        $this->extend('updateEditLinkForDataObject', $link, $obj);
```

Comment:
How come this new method gets an extension hook and the others don't?

Reply:
This one is added to support the additional work in https://github.com/silverstripe/silverstripe-admin/pull/1361 for making CMS edit links easier to deal with - it allows developers to modify the edit link returned by a `ModelAdmin` that isn't directly part of their codebase.

I didn't add one to the others because I didn't have a specific use-case for such an extension in mind - and they feel more like the `Link()` method, which itself doesn't have a hook.

Reply:
OK cool, leave this one in